src/common/commonutils/SshUtils.c

```diff
@@ -339,37 +431,127 @@ int CheckSshClientAliveInterval(char** reason, void* log)
-        OsConfigCaptureReason(reason, "'logingracetime' is not set to a value of 60 or less in SSH Server response (but to %d)",
-            "%s, also 'logingracetime' is not set to a value of 60 or less in SSH Server response (but to %d)", actualValue);
+        OsConfigLogError(log, "CheckSshLoginGraceTime: 'logingracetime' is not set to %d or less in SSH Server response (but to %d)", targetValue, actualValue);
+        OsConfigCaptureReason(reason, "'logingracetime' is not set to a value of %d or less in SSH Server response (but to %d)",
+            "%s, also 'logingracetime' is not set to a value of 60 or less in SSH Server response (but to %d)", targetValue, actualValue);
         status = ENOENT;
     }
     else if (reason)
     {
         FREE_MEMORY(*reason);
-        *reason = FormatAllocateString("%sThe %s service reports that '%s' is set to '%d' (that is 60 or less)", 
-            SECURITY_AUDIT_PASS, g_sshServerService, loginGraceTime, actualValue);
+        *reason = FormatAllocateString("%sThe %s service reports that '%s' is set to '%d' (that is %d or less)", 
+            SECURITY_AUDIT_PASS, g_sshServerService, loginGraceTime, targetValue, actualValue);
     }
 
+    FREE_MEMORY(loginGraceTime);
+
     OsConfigLogInfo(log, "CheckSshLoginGraceTime: %s (%d)", PLAIN_STATUS_FROM_ERRNO(status), status);
 
     return status;
 }
 
-int SetSshOption(const char* option, const char* value, void* log)
+static int CheckSshWarningBanner(const char* bannerFile, const char* bannerText, char** reason, void* log)
+{
+    char* banner = DuplicateStringToLowercase(g_sshBanner);
+    char* actualValue = NULL;
+    char* contents = NULL;
+    int status = 0;
+
+    if ((NULL == bannerFile) || (NULL == bannerText))
+    {
+        OsConfigLogError(log, "CheckSshWarningBanner: invalid arguments");
+        return EINVAL;
+    }
+
+    if (0 == (status = CheckSshOptionIsSet(banner, bannerFile, &actualValue, reason, log)))
+    {
+        if (NULL == (contents = LoadStringFromFile(bannerFile, false, log)))
+        {
+            OsConfigLogError(log, "CheckSshWarningBanner: cannot read from '%s'", bannerFile);
+            OsConfigCaptureReason(reason, "'%s' is set to '%s' but the file cannot be read",
+                "%s, also '%s' is set to '%s' but the file cannot be read", banner, actualValue);
+            status = ENOENT;
+        }
+        else  if (0 != strcmp(contents, bannerText))
+        {
+            OsConfigLogError(log, "CheckSshWarningBanner: banner text is:\n%s instead of:\n%s", contents, bannerText);
+            OsConfigCaptureReason(reason, "Banner text from file '%s'is different from not the expected text",
+                "%s, also the banner text from '%s' is different from the expected text", bannerFile);
+            status = ENOENT;
+        }
+    }
+
+    if ((0 == status) && reason)

```

Comment:
nit: Often this code is explicit when checking for null with `NULL != reason`.

Reply:
True. There is no absolutely consistent use for this throughout the code. The two statements are identical for pointers variables such as this and this allows us to do it brief, like here.  Thanks for the feedback.